### PR TITLE
Add a Kotlin sample. #1880

### PR DESCRIPTION
**Pull request by cnevinc**
This commit transfer `spring-boot-minimal` into Kotlin.

### CODE REVIEW

**Comment by ikhoon** [comment]
@cnevinc  Thanks for contributing! I'm happy to review this PR. 😊
And could you sign for CLA?

File: examples/spring-boot-kotlin/build.gradle [hunk 1]
```diff
@@ -0,0 +1,46 @@
+buildscript {
+    ext.kotlin_version = '1.3.50'
```
Comment (trustin) [inline examples/spring-boot-kotlin/build.gradle]: How about adding `kotlin-stdlib-jdk8` to `dependencies.yml` and refer to the version via `managedVersion`?

File: settings.gradle [hunk 2]
```diff
@@ -55,6 +55,7 @@ includeWithFlags ':examples:grpc-service',           'java'
 includeWithFlags ':examples:proxy-server',           'java'
 includeWithFlags ':examples:saml-service-provider',  'java'
 includeWithFlags ':examples:server-sent-events',     'java'
+includeWithFlags ':examples:spring-boot-kotlin',     'java'
```
Comment (trustin) [inline settings.gradle]: `spring-boot-minimal-kotlin`?

**Comment by trustin** [comment]
Could you also use the different package name with the `spring-boot-minimal` example? It seems like JaCoCo is confused. How about just add `.kt` to the package names?

File: examples/spring-boot-kotlin/src/main/java/example/springframework/boot/minimal/Main.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package example.springframework.boot.minimal
+
+import org.springframework.boot.SpringApplication
+import org.springframework.boot.autoconfigure.SpringBootApplication
+
+@SpringBootApplication
+open class Main {
```
Comment (heowc) [inline examples/spring-boot-kotlin/src/main/java/example/springframework/boot/minimal/Main.kt]: Spring is guiding you to [a better example](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-kotlin.html#boot-features-kotlin-api-runapplication) than the change from java to kotlin function provided by idea.

Wouldn't this be more familiar to others?

**Comment by heowc** [comment]
Hello, Welcome to contribute code to Armeria! :smile:

Although I don't know much about Kotlin, I saw the code and would like to suggest it.

Rather than just converting, I think that using the generator provided by [`start.spring.io`](https://start.spring.io) will be an easier and more user-friendly example.

**Comment by cnevinc** [comment]
> Rather than just converting, I think that using the generator provided by [`start.spring.io`](https://start.spring.io) will be an easier and more user-friendly example.

Good idea. I'll come up with a better example in the other PR. Thank you!

**Comment by trustin** [comment]
@cnevinc Any news on a follow-up? :wink:

**Comment by heowc** [comment]
@trustin  @cnevinc May I challenge this?

**Comment by trustin** [comment]
@heowc Of course! Thanks in advance! :pray:

**Pull request closed without merging**